db/disk/submission.go

```diff
@@ -294,3 +294,30 @@ func (this *backend) getMostRecentSubmissionID(assignment *model.Assignment, ema
 
     return dirents[len(dirents) - 1].Name(), nil;
 }
+
+func (this *backend) RemoveSubmission(assignment *model.Assignment, email string, shortSubmissionID string) (bool, error) {
+    var err error;
+
+    if (shortSubmissionID == "") {
+        shortSubmissionID, err = this.getMostRecentSubmissionID(assignment, email);
+        if (err != nil){
+            return false, fmt.Errorf("Failed to get most recent submission id: `%w`.", err);
+        }
+    }
+
+    if (shortSubmissionID == ""){
+        return false, nil;
+    }
+
+    submissionDir := this.getSubmissionDirFromAssignment(assignment, email, shortSubmissionID);
+
+    if (!util.PathExists(submissionDir)) {
+        return false, fmt.Errorf("Submission does not exist.");
+    }
```

Comment:
Missing a newline after this block.

If this was intentional, then you will want to make sure to use vertical whitespace to separate blocks into chunks of code that does one discrete thing.